eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -1588,7 +1589,47 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
     // peer doesn't cancel the timer
     case Event(TickChannelOpenTimeout, _) => stay()
 
-    // we declare WatchFundingSpentTriggered handlers here because they apply to variants of each state in OFFLINE/SYNCING
+    case Event(w: WatchPublishedTriggered, d: PersistentChannelData) =>
```

Comment:
That doesn't work though, does it? If we were waiting for that event we were in `WAIT_FOR_DUAL_FUNDING_CONFIRMED`. If we received that event while connected, we would go to `WAIT_FOR_DUAL_FUNDING_READY`. But now if that event fires while offline, when we reconnect we will stay stuck in the `WAIT_FOR_DUAL_FUNDING_CONFIRMED` state, won't we?

Same thing for the `WatchFundingConfirmedTriggered` event below...

The existing tests send the `WatchConfirmedTriggered` event after nodes have been reconnected, which hides the issue, but if you add a test that sends this event while offline, it should showcase the bug.

This should be easily fixed by keeping the `channel_reestablish` code that calls `watchFundingConfirmed` to reset these watches and instantly trigger the transition (the unit tests should verify that we reset those watches).

Reply:
🤦‍♂️ embarrassing.

My intent was to simply change the data here. It would be a silent transition.

Reply:
Fixed with 4a12c7255.